workspaces/src/types/mod.rs

```diff
@@ -528,3 +521,26 @@ impl From<Finality> for near_primitives::types::BlockReference {
         value.into()
     }
 }
+
+/// A helper function to create a signed transaction with the given actions.
+pub(crate) fn signed_transaction(
+    signer: &InMemorySigner,
+    actions: Vec<Action>,
+    nonce: near_primitives::types::Nonce,
+    block_hash: near_primitives::hash::CryptoHash,
+    receiver_id: &AccountId,
+) -> SignedTransaction {
+    let tx = Transaction {
+        signer_id: signer.account_id.clone(),
+        public_key: signer.secret_key.public_key().0,
+        nonce,
+        receiver_id: receiver_id.clone(),
+        block_hash: near_primitives::hash::CryptoHash(block_hash.0),
+        actions,
+    };
+
+    SignedTransaction::new(
+        signer.secret_key.0.sign(tx.get_hash_and_size().0.as_ref()),
+        tx,
+    )
```

Comment:
you shouldn't need to call into `SignedTransaction::new`. A `tx.sign(&signer)` is simpler